lib/Common/Memory/ArenaAllocator.h

```diff
@@ -944,21 +944,19 @@ class ReferencedArenaAdapter : public RefCounted
         {
             HeapDelete(this->arena);
         }
-        DeleteCriticalSection(&adapterLock);
     }
 
     ReferencedArenaAdapter(ArenaAllocator* _arena)
         : RefCounted(),
           strongRefCount(0),
           arena(_arena),
           deleteFlag(false)
-    {
-        InitializeCriticalSection(&adapterLock);
-    }
+    { }
 
     bool AddStrongReference()
     {
-        EnterCriticalSection(&adapterLock);
+        bool retval = false;
+        adapterLock.Enter();
```

Comment:
Your change is fine; I was just saying we could use "AutoCriticalSection a(&adapterLock)" for some code like this and the code would be simpler.